js/PageNavModel.js

```diff
@@ -0,0 +1,231 @@
+      Add each combined item to an array
+    */
+    const buttonTypeModels = this.getButtonTypeModels();
+    const currentPageComplete = buttonTypeModels._page.get('_isComplete');
+    const unsortedItems = [];
+    let order = 0;
+    let item;
+
+    for (const type in buttons) {
+      const buttonConfig = buttons[type];
+      let buttonModel = buttonTypeModels[type];
+
+      // Generate sibling entries
+      if (type === '_sibling') {
+        // Skip if only one sibling
+        if (buttonModel.length <= 1) continue;
+
+        buttonModel.forEach((model, index) => {
+          item = model.toJSON();
+          Object.assign(item, buttonConfig, {
+            type,
+            index,
+            _isCurrent: item._id === location._currentId,
+            order: order++,
+            _tooltipId: `pagenav_btn${type}-${index}`,
+            locked: item._isLocked || (buttonConfig._lockUntilPageComplete && !currentPageComplete)
+          });
+          unsortedItems.push(item);
+        });
+
+        continue;
+      }
+
+      // Find buttonModel from config._customRouteId if not found in defined type
+      if (buttonConfig._customRouteId) {
+        buttonModel = data.findById(buttonConfig._customRouteId);
+      }
+
+      // Convert found buttonModel to json if exists or create an 'undefined' json
+      item = buttonModel ? buttonModel.toJSON() : { _isHidden: true };
+
+      Object.assign(item, buttonConfig, {
+        type,
+        index: 0,
+        order: order++,
+        _tooltipId: `pagenav_btn${type}`,
+        locked: item._isLocked || (buttonConfig._lockUntilPageComplete && !currentPageComplete)
+      });
+      unsortedItems.push(item);
+    }
+
+    // requires a stable sorting algorithm - native sorting in Chrome is unstable (should be stable from Chrome 70)
+    const sortedItems = _.sortBy(unsortedItems, '_order');
+
+    sortedItems.forEach((item, index) => {
+      item._index = index;
+    });
+
+    return sortedItems;
+  };
+
+  getButtonTypeModels() {
+    return {
+      _returnToPreviousLocation: this.getReturnToPreviousLocation(),
+      _page: this.getCurrentPage(),
+      _up: this.getCurrentMenu(),
+      _root: Adapt.course,
+      _next: this.getNextPage(),
+      _previous: this.getPrevPage(),
+      _sibling: this.getSiblingPages(),
+      _close: this.getClose()
+    };
+  }
+
+  getReturnToPreviousLocation() {
+    if (!location._previousId) { return; }
+
+    return data.findById(location._previousId);
+  };
+
+  getCurrentPage() {
+    const currentModel = location._currentModel;
+    if (!currentModel.get('_type') === 'page') return;
+
+    return currentModel;
+  };
+
+  getCurrentMenu() {
+    const parents = this.getAncestorModels ? this.getAncestorModels() : this.getParents().models;
+
+    for (const model of parents) {
+      const type = model.get('_type');
+      if (['menu', 'course'].includes(type)) {
+        return model;
+      }
+    }
+  };
```

Comment:
```suggestion
  getCurrentMenu() {
    return this.findAncestor('menu');
  };
```
As the `CourseModel` belongs to the content types: `menu`, `course` and `contentobject` now, this can be shortened.

The classes now have "type groups" they belong to, rather than having to rely on the model's _type data attribute.